coderd/templateversions.go

```diff
@@ -1742,3 +1818,34 @@ func (api *API) publishTemplateUpdate(ctx context.Context, templateID uuid.UUID)
 			slog.F("template_id", templateID), slog.Error(err))
 	}
 }
+
+func checkProvisioners(ctx context.Context, store database.Store, orgID uuid.UUID, wantTags map[string]string, pollInterval time.Duration) (codersdk.MatchedProvisioners, error) {
+	// Check for eligible provisioners. This allows us to return a warning to the user if they
+	// submit a job for which no provisioner is available.
+	eligibleProvisioners, err := store.GetProvisionerDaemonsByOrganization(ctx, database.GetProvisionerDaemonsByOrganizationParams{
+		OrganizationID: orgID,
+		WantTags:       wantTags,
+	})
+	if err != nil {
+		// Log the error but do not return any warnings. This is purely advisory and we should not block.
+		return codersdk.MatchedProvisioners{}, xerrors.Errorf("provisioner daemons by organization: %w", err)
+	}
+
+	threePollsAgo := time.Now().Add(-3 * pollInterval)
```

Comment:
review: this is duplicated from / in line with `coderd/healthcheck`:

```
	if opts.StaleInterval == 0 {
		opts.StaleInterval = provisionerdserver.DefaultHeartbeatInterval * 3
	}
```

Reply:
Do we need the above as a code comment or perhaps a test that checks that they remain in sync?

Reply:
I'm not sure of the value of adding a test for this, but we can possibly refactor where it's defined in a follow-up so that it's consistent. For now, I'll add a comment 👍

Reply:
On the surface it kinda makes sense to use the same logic (3*poll), but at the same time I feel multiplying by 3 is pretty extreme, I'm more in favor of a fixed time like poll+30s or something like that. In this instance poll+30s would surface the issue in half the time. If poll is ever raised, to say 2min, it would go from 1.5min longer to 3.5min, etc.